chia/cmds/cmd_classes.py

```diff
@@ -0,0 +1,311 @@
+        for member in self.my_members.values():
+            all_option_decorators.extend(member.get_all_option_decorators())
+        return all_option_decorators
+
+    def initialize_instance(self, **kwargs: Any) -> ChiaCommand:
+        kwargs_to_pass: Dict[str, Any] = {}
+        for kwarg_name in self.my_kwarg_names:
+            kwargs_to_pass[kwarg_name] = kwargs[kwarg_name]
+
+        for member_arg_name, member in self.my_members.items():
+            kwargs_to_pass[member_arg_name] = member.initialize_instance(**kwargs)
+
+        return self.my_dataclass(**kwargs_to_pass)
+
+    def apply_decorators(self, cmd: SyncCmd) -> SyncCmd:
+        cmd_to_return = cmd
+        if self.needs_context():
+
+            def strip_click_context(func: SyncCmd) -> SyncCmd:
+                def _inner(ctx: click.Context, **kwargs: Any) -> None:
+                    context: Dict[str, Any] = ctx.obj if ctx.obj is not None else {}
+                    func(context=context, **kwargs)
+
+                return _inner
+
+            cmd_to_return = click.pass_context(strip_click_context(cmd_to_return))
+
+        for decorator in self.get_all_option_decorators():
+            cmd_to_return = decorator(cmd_to_return)
+
+        return cmd_to_return
+
+    def __call__(self, *args: Any, **kwargs: Any) -> None:
+        instance = self.initialize_instance(*args, **kwargs)
+        if inspect.iscoroutinefunction(self.my_dataclass.run):
+            coro = instance.run()
+            assert coro is not None
+            asyncio.run(coro)
+        else:
+            instance.run()
+
+
+def _generate_command_parser(cls: Type[ChiaCommand]) -> _CommandParsingStage:
+    option_decorators: List[Callable[[SyncCmd], SyncCmd]] = []
+    kwarg_names: List[str] = []
+    members: Dict[str, _CommandParsingStage] = {}
+    needs_context: bool = False
+
+    hints = get_type_hints(cls)
+    _fields = fields(cls)  # type: ignore[arg-type]
+
+    for _field in _fields:
+        field_name = _field.name
+        if getattr(hints[field_name], COMMAND_HELPER_ATTRIBUTE_NAME, False):
+            members[field_name] = _generate_command_parser(hints[field_name])
+        elif field_name == "context":
+            if hints[field_name] != Context:
+                raise ValueError("only Context can be the hint for variables named 'context'")
+            else:
+                needs_context = True
+                kwarg_names.append(field_name)
+        elif "option_args" in _field.metadata:
+            option_args: Dict[str, Any] = {"multiple": False}
+            option_args.update(_field.metadata["option_args"])
+
+            if "type" not in option_args:
+                origin = get_origin(hints[field_name])
+                if origin == collections.abc.Sequence:
+                    if not option_args["multiple"]:
+                        raise TypeError("Can only use Sequence with multiple=True")
```

Comment:
Is it not possible to just ban specification of `multiple=` and deduce it in all cases from the hint?

Reply:
Like I said in a comment above, I want to preserve everything a dev knows about the ergonomics `click.option` decorators so this function tries to take a "validation" approach rather than inventing new ergonomics that a developer has to learn.  It seems relatively magic to have to know that if you specify `Sequence`, that's the only way to get `multiple=True`.